### PR TITLE
fix(agent-core-v2): redact Unicode, UNC, and C:/ paths in telemetry

### LINKED ISSUE #2418: Telemetry path redaction fails for non-ASCII home directories, UNC paths, and `C:/` spellings
### What happens

`cleanTelemetryString` in `packages/agent-core-v2/src/app/telemetry/privacy.ts` is the last step before an event leaves the process — `CloudAppender.track` cleans properties, buffers the `EnrichedCloudEvent`, and `CloudTransport` POSTs it (`cloudAppender.ts:115`). Its job is to turn absolute file paths into `<REDACTED: user-file-path>`.

It does that for `C:\Users\alice\proj\x.txt` and `/home/alice/proj/x.txt`. It does not do it for a home directory whose name is not ASCII, nor for a UNC path, nor for a drive-letter path written with forward slashes. In those cases the user name — and usually the rest of the path — goes out verbatim.

Both patterns are built from `\w`:

```ts
const POSIX_PATH = /(?:\/[\w.~+-]+){2,}\/?/g;
const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
```

`\w` is `[A-Za-z0-9_]`. Without the `u` flag there is no Unicode property escape, so the match ends at the first non-ASCII byte.

### Measured, on `main`

| case | input | `main` produces | leaked |
| --- | --- | --- | --- |
| POSIX, CJK home dir | `/home/李明/proj/secret.txt` | `/home/李明<REDACTED: user-file-path>` | `李明` |
| Windows, CJK home dir | `C:\Users\李明\proj\secret.txt` | `<REDACTED: user-file-path>李明\proj\secret.txt` | `李明`, `secret.txt` |
| POSIX, Cyrillic home dir | `/home/иван/proj/secret.txt` | `/home/иван<REDACTED: user-file-path>` | `иван` |
| POSIX, accented home dir | `/home/josé/proj/secret.txt` | `<REDACTED: user-file-path>é<REDACTED: user-file-path>` | (name split, `é` emitted) |
| Windows, accented home dir | `C:\Users\josé\proj\secret.txt` | `<REDACTED: user-file-path>é\proj\secret.txt` | `secret.txt` |
| Windows, apostrophe in name | `C:\Users\O'Brien\proj\secret.txt` | `<REDACTED: user-file-path>'Brien\proj\secret.txt` | `Brien`, `secret.txt` |
| UNC share | `\\fileserver\home\alice.chen\proj\secret.txt` | *(unchanged)* | the whole path |
| Windows, forward slashes | `C:/Users/alice.chen/proj/secret.txt` | `C:<REDACTED: user-file-path>` | drive letter |

Three separate root causes:

1. **`\w` is ASCII-only.** Any non-ASCII character in any segment truncates the match there. A CJK, Cyrillic, or accented home directory name is the common case; so is `O'Brien`, since `'` is not in `\w` either.
2. **UNC and `\\?\` long-path forms are not matched at all.** `WINDOWS_PATH` requires `[A-Za-z]:\`, and `\b` before it does not help — `\\server\share\...` has no drive letter, so nothing matches and the value is emitted as-is.
3. **`C:/a/b` is only half-handled.** `WINDOWS_PATH` requires a backslash, so the drive letter is left behind and the POSIX pattern redacts only the tail, yielding `C:<REDACTED: ...>`. Node normalizes to forward slashes in plenty of places, so this spelling reaches telemetry in practice.

### Two smaller behaviour bugs in the same function

- **The `node_modules/` tail is discarded on Windows.** The marker is spelled with a forward slash and the Windows branch does not look for it at all, so `C:\...\node_modules\pkg\index.js` becomes a bare placeholder. The module docstring says these tails are kept "because they carry diagnostic value without user data" — that only holds on POSIX today.
- **`WINDOWS_PATH` allows a bare space inside a segment with no separator required after it**, so `C:\Program Files\a.txt could not be read` redacts to `<REDACTED: user-file-path>` — the message text is swallowed. Not a privacy leak, but it destroys the diagnostic.

### Why it matters

This is the final hop before a remote POST, and the redaction is silent: nothing logs or errors when a pattern fails to match, so a leak looks identical to a clean event. `agent-core-v2` is consumed by `apps/kimi-code` and several other packages, so it applies to shipped builds.

Non-ASCII home directory names are not an edge case for this project's user base in particular. On a Chinese-locale Windows install, `C:\Users\<name>` with a CJK name is ordinary.

There is currently no test file for `privacy.ts`, so none of this is covered.

### Expected

Every absolute path is redacted regardless of the script its segments are written in, and regardless of which Windows spelling it uses; the `node_modules/` tail is preserved on both platforms; text that is not an absolute path is left alone.

### Environment

`main`, `packages/agent-core-v2`. Reproducible by calling `cleanTelemetryString` directly with the inputs above.

### PR DESCRIPTION
**Pull request by mangeshraut712**
## Related Issue

Resolve #2418

## Problem

See linked issue. `cleanTelemetryString` failed to fully redact absolute paths when a home directory was non-ASCII, when the path was UNC / `\\?\` long-path, or when a Windows drive path used forward slashes — leaking usernames (and often the rest of the path) into outbound telemetry. Windows `node_modules` tails were also dropped, and spaced paths like `C:\Program Files\…` could swallow trailing diagnostic text.

## What changed

- Widened path matching in `privacy.ts` to Unicode letter/number segments (`\p{L}` / `\p{N}`), UNC and long-path prefixes, and `C:/…` spellings.
- Keep `node_modules/…` tails on both platforms; stop Windows matches before trailing message text after a spaced directory.
- Added unit coverage for the cases from #2418.

## Checklist

- [x] I have read the CONTRIBUTING document.
- [x] I have linked a related issue, or explained the problem above.
- [x] I have added tests that prove my feature works.
- [x] Ran `gen-changesets` skill, or this PR needs no changeset.
- [x] Ran `gen-docs` skill, or this PR needs no doc update.

### CODE REVIEW

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 1]
```diff
@@ -21,19 +26,41 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
   [/\b(?:sk|pk|ak)-[A-Za-z0-9_-]{16,}\b/g, '<REDACTED: API Key>'],
 ];
 
-const POSIX_PATH = /(?:\/[\w.~+-]+){2,}\/?/g;
-const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
+/** Path segment atom: letters/digits (any script) plus common filename marks — no spaces. */
+const SEGMENT_ATOM = String.raw`[\p{L}\p{N}._'~+-]+`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Addressed in 07df487f and rebased in e049686f: path atoms now include Unicode combining marks; spaced final filenames are fully redacted while diagnostic text remains; trailing spaced directories are covered; implementation comments were removed under current conventions. Post-rebase telemetry suite: 53 tests pass, agent-core-v2 typecheck and oxlint are clean.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 2]
```diff
@@ -21,19 +26,41 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
...
+const WINDOWS_DRIVE = String.raw`\b[A-Za-z]:(?:[\\/]${SEGMENT_DIR})+[\\/]${SEGMENT_ATOM}`;
+/** POSIX absolute paths with at least two segments (`/home/alice/…`). */
+const POSIX_PATH = String.raw`(?:\/${SEGMENT_DIR})*\/${SEGMENT_ATOM}(?:\/${SEGMENT_ATOM})+\/?`;
+
+/**
+ * Absolute paths in one alternation (Windows first) so a replacement — including a
+ * kept `node_modules/…` tail — is not re-matched by the POSIX arm.
+ *
+ * Intermediate segments may include spaces; the final segment may not, so
+ * trailing message text after a path is preserved.
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Addressed in 07df487f and rebased in e049686f: path atoms now include Unicode combining marks; spaced final filenames are fully redacted while diagnostic text remains; trailing spaced directories are covered; implementation comments were removed under current conventions. Post-rebase telemetry suite: 53 tests pass, agent-core-v2 typecheck and oxlint are clean.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 3]
```diff
@@ -21,19 +26,41 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
   [/\b(?:sk|pk|ak)-[A-Za-z0-9_-]{16,}\b/g, '<REDACTED: API Key>'],
 ];
 
-const POSIX_PATH = /(?:\/[\w.~+-]+){2,}\/?/g;
-const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
+/** Path segment atom: letters/digits (any script) plus common filename marks — no spaces. */
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Addressed in 07df487f and rebased in e049686f: path atoms now include Unicode combining marks; spaced final filenames are fully redacted while diagnostic text remains; trailing spaced directories are covered; implementation comments were removed under current conventions. Post-rebase telemetry suite: 53 tests pass, agent-core-v2 typecheck and oxlint are clean.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 4]
```diff
@@ -11,19 +11,30 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
   [/\b(?:sk|pk|ak)-[A-Za-z0-9_-]{16,}\b/g, '<REDACTED: API Key>'],
 ];
 
-const POSIX_PATH = /(?:\/[\w.~+-]+){2,}\/?/g;
-const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
+const SEGMENT_ATOM = String.raw`[\p{L}\p{M}\p{N}._'~+-]+`;
+const SEGMENT_DIR = String.raw`${SEGMENT_ATOM}(?: ${SEGMENT_ATOM})*`;
+const FINAL_WITH_EXTENSION = String.raw`${SEGMENT_ATOM}(?: ${SEGMENT_ATOM})*?\.[\p{L}\p{M}\p{N}]{1,16}`;
+const FINAL_SEGMENT = String.raw`(?:${FINAL_WITH_EXTENSION}|${SEGMENT_ATOM})`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in 608f1a9a: FINAL_SEGMENT now accepts extensionless spaced components, while extension-bearing paths still stop before diagnostic text; UNC and long-UNC share names use the same spaced-segment grammar. Added Windows/POSIX extensionless and UNC shared-name regressions. Verification: 54 telemetry tests, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 5]
```diff
@@ -11,19 +11,30 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
...
+const WINDOWS_TAIL = String.raw`(?:(?:[\\/]${SEGMENT_DIR})+[\\/]|(?:[\\/]${SEGMENT_DIR})*[\\/]${FINAL_SEGMENT})`;
+const PATH_BOUNDARY = String.raw`(?![\p{L}\p{M}\p{N}._'~+\\/-])`;
+
+const WINDOWS_UNC_OR_LONG = String.raw`(?:\\\\\?(?:\\(?:UNC\\[^\s\\/]+\\[^\s\\/]+|[A-Za-z]:))|\\\\[^\s\\/]+\\[^\s\\/]+)${WINDOWS_TAIL}${PATH_BOUNDARY}`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in 608f1a9a: FINAL_SEGMENT now accepts extensionless spaced components, while extension-bearing paths still stop before diagnostic text; UNC and long-UNC share names use the same spaced-segment grammar. Added Windows/POSIX extensionless and UNC shared-name regressions. Verification: 54 telemetry tests, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 6]
```diff
@@ -11,19 +11,30 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
...
-const POSIX_PATH = /(?:\/[\w.~+-]+){2,}\/?/g;
-const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
+const SEGMENT_ATOM = String.raw`[\p{L}\p{M}\p{N}._'~+-]+`;
+const SEGMENT_DIR = String.raw`${SEGMENT_ATOM}(?: ${SEGMENT_ATOM})*`;
+const FINAL_WITH_EXTENSION = String.raw`${SEGMENT_ATOM}(?: ${SEGMENT_ATOM})*?\.[\p{L}\p{M}\p{N}]{1,16}`;
+const FINAL_SEGMENT = String.raw`(?:${FINAL_WITH_EXTENSION}|${SEGMENT_DIR})`;
+const WINDOWS_TAIL = String.raw`(?:(?:[\\/]${SEGMENT_DIR})+[\\/]|(?:[\\/]${SEGMENT_DIR})*[\\/]${FINAL_SEGMENT})`;
+const PATH_BOUNDARY = String.raw`(?![\p{L}\p{M}\p{N}._'~+\\/-])`;
+
+const WINDOWS_UNC_OR_LONG = String.raw`(?:\\\\\?(?:\\(?:UNC\\[^\s\\/]+\\${SEGMENT_DIR}|[A-Za-z]:))|\\\\[^\s\\/]+\\${SEGMENT_DIR})${WINDOWS_TAIL}${PATH_BOUNDARY}`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in dedfc701: UNC server/share roots are now redacted with or without a trailing separator; path segments accept repeated spaces; apostrophes are allowed only internally between path atoms so closing single quotes terminate the match and preserve diagnostic suffixes. Added regressions for all three forms. Telemetry suite 54/54, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 7]
```diff
@@ -11,19 +11,30 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
   [/\b(?:sk|pk|ak)-[A-Za-z0-9_-]{16,}\b/g, '<REDACTED: API Key>'],
 ];
 
-const POSIX_PATH = /(?:\/[\w.~+-]+){2,}\/?/g;
-const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
+const SEGMENT_ATOM = String.raw`[\p{L}\p{M}\p{N}._'~+-]+`;
+const SEGMENT_DIR = String.raw`${SEGMENT_ATOM}(?: ${SEGMENT_ATOM})*`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in dedfc701: UNC server/share roots are now redacted with or without a trailing separator; path segments accept repeated spaces; apostrophes are allowed only internally between path atoms so closing single quotes terminate the match and preserve diagnostic suffixes. Added regressions for all three forms. Telemetry suite 54/54, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 8]
```diff
@@ -11,19 +11,30 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
...
+const FINAL_WITH_EXTENSION = String.raw`${SEGMENT_ATOM}(?: ${SEGMENT_ATOM})*?\.[\p{L}\p{M}\p{N}]{1,16}`;
+const FINAL_SEGMENT = String.raw`(?:${FINAL_WITH_EXTENSION}|${SEGMENT_DIR})`;
+const WINDOWS_TAIL = String.raw`(?:(?:[\\/]${SEGMENT_DIR})+[\\/]|(?:[\\/]${SEGMENT_DIR})*[\\/]${FINAL_SEGMENT})`;
+const PATH_BOUNDARY = String.raw`(?![\p{L}\p{M}\p{N}._'~+\\/-])`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in dedfc701: UNC server/share roots are now redacted with or without a trailing separator; path segments accept repeated spaces; apostrophes are allowed only internally between path atoms so closing single quotes terminate the match and preserve diagnostic suffixes. Added regressions for all three forms. Telemetry suite 54/54, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 9]
```diff
@@ -11,19 +11,32 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
...
-const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
+const SEGMENT_CORE = String.raw`[\p{L}\p{M}\p{N}._~+-]+`;
+const SEGMENT_ATOM = String.raw`${SEGMENT_CORE}(?:'${SEGMENT_CORE})*`;
+const SEGMENT_DIR = String.raw`${SEGMENT_ATOM}(?: +${SEGMENT_ATOM})*`;
+const FINAL_WITH_EXTENSION = String.raw`${SEGMENT_ATOM}(?: +${SEGMENT_ATOM})*?\.[\p{L}\p{M}\p{N}]{1,16}`;
+const FINAL_SEGMENT = String.raw`(?:${FINAL_WITH_EXTENSION}|${SEGMENT_DIR})`;
+const WINDOWS_TAIL = String.raw`(?:(?:[\\/]${SEGMENT_DIR})+[\\/]|(?:[\\/]${SEGMENT_DIR})*[\\/]${FINAL_SEGMENT})`;
+const PATH_BOUNDARY = String.raw`(?![\p{L}\p{M}\p{N}._~+\\/-])`;
+
+const WINDOWS_UNC_BASE = String.raw`(?:\\\\\?(?:\\(?:UNC\\[^\s\\/]+\\${SEGMENT_DIR}|[A-Za-z]:))|\\\\[^\s\\/]+\\${SEGMENT_DIR})`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in f3bde11d: path atoms now include dollar signs and the boundary rejects partial admin-share matches; extended-length paths use a dedicated component grammar that permits trailing spaces without broadening ordinary diagnostic-text matching. Added C$ and trailing-space extended-path regressions. Telemetry suite 54/54, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 10]
```diff
@@ -11,19 +11,32 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
   [/\b(?:sk|pk|ak)-[A-Za-z0-9_-]{16,}\b/g, '<REDACTED: API Key>'],
 ];
 
-const POSIX_PATH = /(?:\/[\w.~+-]+){2,}\/?/g;
-const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
+const SEGMENT_CORE = String.raw`[\p{L}\p{M}\p{N}._~+-]+`;
+const SEGMENT_ATOM = String.raw`${SEGMENT_CORE}(?:'${SEGMENT_CORE})*`;
+const SEGMENT_DIR = String.raw`${SEGMENT_ATOM}(?: +${SEGMENT_ATOM})*`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in f3bde11d: path atoms now include dollar signs and the boundary rejects partial admin-share matches; extended-length paths use a dedicated component grammar that permits trailing spaces without broadening ordinary diagnostic-text matching. Added C$ and trailing-space extended-path regressions. Telemetry suite 54/54, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 11]
```diff
@@ -11,19 +11,35 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
...
+const FINAL_WITH_EXTENSION = String.raw`${SEGMENT_ATOM}(?: +${SEGMENT_ATOM})*?\.[\p{L}\p{M}\p{N}]{1,16}`;
+const FINAL_SEGMENT = String.raw`(?:${FINAL_WITH_EXTENSION}|${SEGMENT_DIR})`;
+const WINDOWS_TAIL = String.raw`(?:(?:[\\/]${SEGMENT_DIR})+[\\/]|(?:[\\/]${SEGMENT_DIR})*[\\/]${FINAL_SEGMENT})`;
+const SEGMENT_DIR_EXTENDED = String.raw`${SEGMENT_DIR} *`;
+const WINDOWS_TAIL_EXTENDED = String.raw`(?:(?:[\\/]${SEGMENT_DIR_EXTENDED})+[\\/]|(?:[\\/]${SEGMENT_DIR_EXTENDED})*[\\/]${FINAL_SEGMENT})`;
+const PATH_BOUNDARY = String.raw`(?![\p{L}\p{M}\p{N}.$_~+\\/-])`;
+
+const WINDOWS_LONG_BASE = String.raw`\\\\\?\\(?:UNC\\[^\s\\/]+\\${SEGMENT_DIR}|[A-Za-z]:)`;
+const WINDOWS_UNC_BASE = String.raw`\\\\[^\s\\/]+\\${SEGMENT_DIR}`;
+const WINDOWS_UNC_OR_LONG = String.raw`(?:${WINDOWS_LONG_BASE}(?:${WINDOWS_TAIL_EXTENDED}|[\\/]?)|${WINDOWS_UNC_BASE}(?:${WINDOWS_TAIL}|[\\/]?))${PATH_BOUNDARY}`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in 1697c0c6: UNC shares now use the Windows legal-character exclusion set, preventing partial matches at #/$ and similar punctuation. node_modules preservation is case-sensitive on POSIX and case-insensitive only for Windows paths. Added team#1 and uppercase POSIX NODE_MODULES regressions. Telemetry suite 54/54, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 12]
```diff
@@ -11,19 +11,35 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
...
+const WINDOWS_DRIVE = String.raw`\b[A-Za-z]:${WINDOWS_TAIL}${PATH_BOUNDARY}`;
+const POSIX_PATH = String.raw`(?:(?:\/${SEGMENT_DIR}){2,}\/|(?:\/${SEGMENT_DIR})+\/${FINAL_SEGMENT})${PATH_BOUNDARY}`;
+const ABSOLUTE_PATH = new RegExp(`${WINDOWS_UNC_OR_LONG}|${WINDOWS_DRIVE}|${POSIX_PATH}`, 'gu');
+
+function redactAbsolutePath(match: string): string {
+  const normalized = match.replaceAll('\\', '/');
+  const index = normalized.toLowerCase().indexOf(NODE_MODULES_MARKER);
+  return index === -1 ? REDACTED_PATH : normalized.slice(index);
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in 1697c0c6: UNC shares now use the Windows legal-character exclusion set, preventing partial matches at #/$ and similar punctuation. node_modules preservation is case-sensitive on POSIX and case-insensitive only for Windows paths. Added team#1 and uppercase POSIX NODE_MODULES regressions. Telemetry suite 54/54, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 13]
```diff
@@ -11,19 +11,39 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
   [/\b(?:sk|pk|ak)-[A-Za-z0-9_-]{16,}\b/g, '<REDACTED: API Key>'],
 ];
 
-const POSIX_PATH = /(?:\/[\w.~+-]+){2,}\/?/g;
-const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
+const SEGMENT_CORE = String.raw`[\p{L}\p{M}\p{N}.$_~+-]+`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in 54c320ef with a structural change: every Windows tail component now uses the legal-character exclusion grammar, so punctuation is handled throughout rather than enumerated; extension-ending paths retain diagnostic suffixes, while generic paths redact conservatively. Strings above 8192 characters are redacted before regex processing, bounding synchronous appender work. Added alice#1 and 60k dotted-input regressions. Telemetry suite 55/55, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 14]
```diff
@@ -11,19 +11,39 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
...
+const SEGMENT_ATOM = String.raw`${SEGMENT_CORE}(?:'${SEGMENT_CORE})*`;
+const SEGMENT_DIR = String.raw`${SEGMENT_ATOM}(?: +${SEGMENT_ATOM})*`;
+const FINAL_WITH_EXTENSION = String.raw`${SEGMENT_ATOM}(?: +${SEGMENT_ATOM})*?\.[\p{L}\p{M}\p{N}]{1,16}`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in 54c320ef with a structural change: every Windows tail component now uses the legal-character exclusion grammar, so punctuation is handled throughout rather than enumerated; extension-ending paths retain diagnostic suffixes, while generic paths redact conservatively. Strings above 8192 characters are redacted before regex processing, bounding synchronous appender work. Added alice#1 and 60k dotted-input regressions. Telemetry suite 55/55, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 15]
```diff
@@ -11,19 +11,41 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
   [/\b(?:sk|pk|ak)-[A-Za-z0-9_-]{16,}\b/g, '<REDACTED: API Key>'],
 ];
 
-const POSIX_PATH = /(?:\/[\w.~+-]+){2,}\/?/g;
-const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
+const SEGMENT_CORE = String.raw`[\p{L}\p{M}\p{N}.$_~+-]+`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in 32bb1052: POSIX components now use the platform rule (any character except slash, NUL, and delimiter quotes), with a separate extension-ending branch to preserve diagnostic suffixes. Added emoji and comma component regressions. Also reran the unrelated flaky MCP stdio test from the prior CI failure: 73/73 focused tests pass; typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 16]
```diff
@@ -11,19 +11,41 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
...
+const POSIX_GENERIC_PATH = String.raw`(?:\/${POSIX_COMPONENT}){2,}\/?(?=$|['"])`;
+const POSIX_PATH = String.raw`(?:${POSIX_EXTENSION_PATH}|${POSIX_GENERIC_PATH})`;
+const ABSOLUTE_PATH = new RegExp(`${WINDOWS_UNC_OR_LONG}|${WINDOWS_DRIVE}|${POSIX_PATH}`, 'gu');
+
+function redactAbsolutePath(match: string): string {
+  const normalized = match.replaceAll('\\', '/');
+  const windowsPath = /^[A-Za-z]:[\\/]|^\\\\/.test(match);
+  const index = windowsPath
+    ? normalized.toLowerCase().indexOf(NODE_MODULES_MARKER)
+    : normalized.indexOf(NODE_MODULES_MARKER);
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts:39]: Fixed in 0e7abb2f: dependency-tail preservation now requires the complete /node_modules/ path component and returns from its component boundary; extension paths recognize common closing punctuation so diagnostic suffixes remain. Added false-marker and Windows/POSIX comma regressions. Focused telemetry + MCP tests 73/73, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 17]
```diff
@@ -11,19 +11,41 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
   [/\b(?:sk|pk|ak)-[A-Za-z0-9_-]{16,}\b/g, '<REDACTED: API Key>'],
 ];
 
-const POSIX_PATH = /(?:\/[\w.~+-]+){2,}\/?/g;
-const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
+const PATH_BOUNDARY = String.raw`(?![\p{L}\p{M}\p{N}.$_~+\\/-])`;
+const WINDOWS_COMPONENT = String.raw`[^\\/:*?"<>|\u0000-\u001F]+`;
+const WINDOWS_FINAL_WITH_EXTENSION = String.raw`${WINDOWS_COMPONENT}?\.[\p{L}\p{M}\p{N}]{1,16}`;
+const WINDOWS_EXTENSION_TAIL = String.raw`(?:[\\/]${WINDOWS_COMPONENT})*[\\/]${WINDOWS_FINAL_WITH_EXTENSION}(?=$|[\s'"])`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in 0e7abb2f: dependency-tail preservation now requires the complete /node_modules/ path component and returns from its component boundary; extension paths recognize common closing punctuation so diagnostic suffixes remain. Added false-marker and Windows/POSIX comma regressions. Focused telemetry + MCP tests 73/73, typecheck, comment guard, and repository lint exit successfully.

File: packages/agent-core-v2/src/app/telemetry/privacy.ts [hunk 18]
```diff
@@ -11,19 +11,42 @@ const LABELED_PATTERNS: ReadonlyArray<readonly [RegExp, string]> = [
...
 ];
 
-const POSIX_PATH = /(?:\/[\w.~+-]+){2,}\/?/g;
-const WINDOWS_PATH = /\b[A-Za-z]:\\(?:[\w.~ -]+\\?){2,}/g;
+const PATH_BOUNDARY = String.raw`(?![\p{L}\p{M}\p{N}.$_~+\\/-])`;
+const EXTENSION_BOUNDARY = String.raw`(?=$|[\s'",;:!?) }])`;
+const WINDOWS_COMPONENT = String.raw`[^\\/:*?"<>|\u0000-\u001F]+`;
+const WINDOWS_FINAL_WITH_EXTENSION = String.raw`${WINDOWS_COMPONENT}?\.[\p{L}\p{M}\p{N}]{1,16}`;
+const WINDOWS_EXTENSION_TAIL = String.raw`(?:[\\/]${WINDOWS_COMPONENT})*[\\/]${WINDOWS_FINAL_WITH_EXTENSION}${EXTENSION_BOUNDARY}`;
+const WINDOWS_GENERIC_TAIL = String.raw`(?:[\\/]${WINDOWS_COMPONENT})+[\\/]?(?=$|['"])`;
```
[the remark this answers is not in the document]
Comment (mangeshraut712) [reply packages/agent-core-v2/src/app/telemetry/privacy.ts]: Fixed in e70cbe8d: generic Windows tails may now terminate at characters Windows excludes from components (including colon and control/newline delimiters), while separators remain disallowed as partial-match boundaries. Added colon and newline extensionless-path regressions. Focused telemetry + MCP tests 73/73, typecheck, comment guard, and repository lint exit successfully.
